File: code/MessageGenerator/security_protocols/dlapp/models/pseudo_identity.py
class PseudoIdentityDLAPP:
    """
    Class representing a PID (Pseudo-Identity)

    Pseudo-identities can hide the real identity of the ITS-S from other ITS stations and prevent tracking attacks
    """

    def __init__(self, byte_array):
        self.byte_array = byte_array

    def __eq__(self, other):
        """
        Check if two PseudoIdentity objects are equal based on their byte_array
        """
        if isinstance(other, PseudoIdentityDLAPP):
            return self.byte_array == other.byte_array
        return False

    def __ne__(self, other):
        """
        Check if two PseudoIdentity objects are not equal based on their byte_array
        """
        return not self.__eq__(other)

    def __str__(self):
        return f"PseudoIdentity(bytes={self.byte_array})"

File: code/Android_App/ITS-Mobile-App/app/src/main/java/pt/isel/meic/tfm/itsmobileapp/security_protocols/mfspv/models/PseudoIdentity.kt
package pt.isel.meic.tfm.itsmobileapp.security_protocols.mfspv.models

/**
 * Class representing a PID (Pseudo-Identity)
 *
 * Pseudo-identities can hide the real identity of the ITS-S from other ITS stations and prevent tracking attacks
 */
class PseudoIdentityMFSPV(
    val bytes : ByteArray
) {
    override fun toString(): String {
        return "PseudoIdentity(bytes=${bytes.contentToString()})"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is PseudoIdentityMFSPV) return false

        return bytes.contentEquals(other.bytes)
    }

    override fun hashCode(): Int {
        return bytes.contentHashCode()
    }

}
